activation/atxdb.go

```diff
@@ -435,7 +447,7 @@ func (db *DB) SyntacticallyValidateAtx(atx *types.ActivationTx) error {
 // ContextuallyValidateAtx ensures that the previous ATX referenced is the last known ATX for the referenced miner ID.
 // If a previous ATX is not referenced, it validates that indeed there's no previous known ATX for that miner ID.
 func (db *DB) ContextuallyValidateAtx(atx *types.ActivationTxHeader) error {
-	if atx.PrevATXID != *types.EmptyATXID {
+	if atx.PrevATXID != db.goldenATXID {
```

Comment:
a suggestion - if you change `GetNodeLastAtxID` to return GoldenATX if no ATX wAs found this flow can be shorter... it might be not an easy switch since this is used elsewhere... so just take a look and think wether it's worth it or not

Reply:
I think `ContextuallyValidateAtx` is a bit complex to understand, so I think it's worth it. I made the suggested changes in the last commit. If we don't need them now, I can create a new branch and remove the last commit from this PR.